netstat/netstat_linux.go

```diff
@@ -212,8 +212,10 @@ func (p *procFd) iterFdDir() {
 				if err != nil {
 					return
 				}
+				if stat != nil {
+					defer stat.Close()
+				}
 				n, err := stat.Read(buf[:])
-				stat.Close()
 				if err != nil {
```

Comment:
This is wrong. You don't need to check for nil there. `os.Open` always returns non-nil
value if it succeeded. Check the source code for os.Open.